File: macapp/KWEngine/UI/Theme.swift
//  Theme.swift
//  KWEngine — design system: Pantone palette, minimal premium Apple style.
//
//  Principles: color is semantic (never decorative), generous whitespace,
//  hairline borders, SF Pro + rounded numerals, full light/dark support.

import SwiftUI
import AppKit

// MARK: - Pantone palette (hex from Pantone color bridge)

extension NSColor {
    convenience init(hex: String) {
        var v = UInt64(0)
        Scanner(string: hex.replacingOccurrences(of: "#", with: "")).scanHexInt64(&v)
        self.init(srgbRed: CGFloat((v >> 16) & 0xFF) / 255,
                  green: CGFloat((v >> 8) & 0xFF) / 255,
                  blue: CGFloat(v & 0xFF) / 255,
                  alpha: 1)
    }
}

extension Color {
    /// Dynamic color that follows the system appearance.
    init(lightHex: String, darkHex: String) {
        self.init(nsColor: NSColor(name: nil) { appearance in
            let isDark = appearance.bestMatch(from: [.darkAqua, .aqua]) == .darkAqua
            return NSColor(hex: isDark ? darkHex : lightHex)
        })
    }
}

enum Theme {

    // Core brand
    /// PANTONE 19-4052 Classic Blue — primary accent.
    static let accent = Color(lightHex: "0F4C81", darkHex: "7AA5CC")
    /// PANTONE 533 C — deep navy, hero surfaces.
    static let navy = Color(lightHex: "1F2A44", darkHex: "27344F")

    // Ink (text)
    /// PANTONE Black 6 C / Bright White 11-0601.
    static let ink = Color(lightHex: "101820", darkHex: "F4F5F0")
    /// PANTONE Cool Gray 11 C / Cool Gray 4 C.
    static let inkSecondary = Color(lightHex: "53565A", darkHex: "A8AAAD")

    // Semantic
    /// PANTONE 17-5641 Emerald — success / distilled.
    static let emerald = Color(lightHex: "009473", darkHex: "2FBF96")
    /// PANTONE 1235 C Saffron — warnings / stale / incomplete.
    static let saffron = Color(lightHex: "E8A317", darkHex: "FFC64A")
    /// PANTONE 485 C Fiery Red — errors.
    static let fieryRed = Color(lightHex: "DA291C", darkHex: "F0564A")
    /// PANTONE 16-1546 Living Coral — Ask / highlights.
    static let coral = Color(lightHex: "E85D4E", darkHex: "FF8A7E")
    /// PANTONE 18-3838 Ultra Violet — math-basis tags.
    static let ultraViolet = Color(lightHex: "5F4B8B", darkHex: "9786BE")
    /// PANTONE 3262 C — data-regime tags.
    static let teal = Color(lightHex: "00897F", darkHex: "34D6C9")

    // Surfaces
    /// Canvas tint: PANTONE 11-0601 Bright White / near-black slate.
    static let canvas = Color(lightHex: "F4F5F0", darkHex: "17191D")
    /// Cards float one step above the canvas.
    static let card = Color(lightHex: "FFFFFF", darkHex: "212429")
    /// Hairline: PANTONE Cool Gray 1 C, translucent.
    static let hairline = Color(lightHex: "D9D9D6", darkHex: "3A3D42").opacity(0.8)
    /// Subtle fill for inset wells (notes, code).
    static let well = Color(lightHex: "EDEEE9", darkHex: "1B1E22")

    static func status(_ s: PaperStatus) -> Color {
        switch s {
        case .pending: return inkSecondary
        case .l1: return accent
        case .l2: return teal
        case .complete: return emerald
        case .incomplete: return saffron
        }
    }

    static func statusLabel(_ s: PaperStatus) -> String {
        switch s {
        case .pending: return tr("待读", "Pending")
        case .l1: return tr("已读 L1", "Read · L1")
        case .l2: return "L2"
        case .complete: return tr("已蒸馏", "Distilled")
        case .incomplete: return tr("不完整", "Incomplete")
        }
    }
}

// MARK: - Reusable styles

/// Uppercased, tracked section label — the quiet Apple-style eyebrow.
struct KWSectionLabel: ViewModifier {
    func body(content: Content) -> some View {
        content
            .font(.system(size: 11, weight: .semibold))
            .tracking(0.8)
            .foregroundColor(Theme.inkSecondary)
            .textCase(.uppercase)
    }
}

/// Card surface: white/dark elevated panel with hairline border.
struct KWCard: ViewModifier {
    var padding: CGFloat = 14
    func body(content: Content) -> some View {
        content
            .padding(padding)
            .background(
                RoundedRectangle(cornerRadius: 10, style: .continuous)
                    .fill(Theme.card)
            )
            .overlay(
                RoundedRectangle(cornerRadius: 10, style: .continuous)
                    .strokeBorder(Theme.hairline, lineWidth: 1)
            )
    }
}

/// Inset well for body notes / long text.
struct KWWell: ViewModifier {
    func body(content: Content) -> some View {
        content
            .padding(12)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(
                RoundedRectangle(cornerRadius: 8, style: .continuous)
                    .fill(Theme.well)
            )
            .overlay(
                RoundedRectangle(cornerRadius: 8, style: .continuous)
                    .strokeBorder(Theme.hairline, lineWidth: 1)
            )
    }
}

extension View {
    func kwSectionLabel() -> some View { modifier(KWSectionLabel()) }
    func kwCard(padding: CGFloat = 14) -> some View { modifier(KWCard(padding: padding)) }
    func kwWell() -> some View { modifier(KWWell()) }
}

/// Capsule tag chip — tinted fill, hairline ring, semantic color.
struct KWChip: View {
    let text: String
    var color: Color = Theme.accent

    var body: some View {
        Text(text)
            .font(.system(size: 12.5, weight: .medium))
            .foregroundColor(color)
            .padding(.horizontal, 10)
            .padding(.vertical, 4)
            .background(Capsule(style: .continuous).fill(color.opacity(0.10)))
            .overlay(Capsule(style: .continuous).strokeBorder(color.opacity(0.28), lineWidth: 1))
            .textSelection(.enabled)
    }
}

/// The brand glyph: three layers distilling to an essence dot.
/// Mirrors the app icon — used on the welcome hero.
struct EngineGlyph: View {
    var tint: Color = .white
    var scale: CGFloat = 1

    var body: some View {
        VStack(spacing: 9 * scale) {
            Capsule().fill(tint.opacity(0.45)).frame(width: 86 * scale, height: 13 * scale)
            Capsule().fill(tint.opacity(0.70)).frame(width: 60 * scale, height: 13 * scale)
            Capsule().fill(tint).frame(width: 36 * scale, height: 13 * scale)
            Circle().fill(tint).frame(width: 13 * scale, height: 13 * scale)
        }
    }
}

File: winapp/src/kwengine_app/theme.py
"""Pantone design system — same palette as the macOS app, in Qt clothing.
Color is semantic, never decorative. Light/dark follow the OS."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class Palette:
    accent: str          # PANTONE 19-4052 Classic Blue
    navy: str            # PANTONE 533 C
    ink: str             # PANTONE Black 6 C / Bright White
    ink_secondary: str   # PANTONE Cool Gray 11 C / 4 C
    emerald: str         # PANTONE 17-5641
    saffron: str         # PANTONE 1235 C
    fiery_red: str       # PANTONE 485 C
    coral: str           # PANTONE 16-1546 Living Coral
    ultra_violet: str    # PANTONE 18-3838
    teal: str            # PANTONE 3262 C
    canvas: str          # PANTONE 11-0601 Bright White / slate
    card: str
    hairline: str
    well: str
    sidebar: str
    is_dark: bool = False


LIGHT = Palette(
    accent="#0F4C81", navy="#1F2A44", ink="#101820", ink_secondary="#53565A",
    emerald="#009473", saffron="#E8A317", fiery_red="#DA291C", coral="#E85D4E",
    ultra_violet="#5F4B8B", teal="#00897F",
    canvas="#F4F5F0", card="#FFFFFF", hairline="#D9D9D6", well="#EDEEE9",
    sidebar="#ECEDE8", is_dark=False,
)

DARK = Palette(
    accent="#7AA5CC", navy="#27344F", ink="#F4F5F0", ink_secondary="#A8AAAD",
    emerald="#2FBF96", saffron="#FFC64A", fiery_red="#F0564A", coral="#FF8A7E",
    ultra_violet="#9786BE", teal="#34D6C9",
    canvas="#17191D", card="#212429", hairline="#3A3D42", well="#1B1E22",
    sidebar="#1D2024", is_dark=True,
)

_current = LIGHT


def detect_palette() -> Palette:
    global _current
    try:
        from PySide6.QtCore import Qt
        from PySide6.QtGui import QGuiApplication
        scheme = QGuiApplication.styleHints().colorScheme()
        _current = DARK if scheme == Qt.ColorScheme.Dark else LIGHT
    except Exception:
        _current = LIGHT
    return _current


def palette() -> Palette:
    return _current


def status_color(p: Palette, status: str) -> str:
    return {
        "pending": p.ink_secondary,
        "L1": p.accent,
        "L2": p.teal,
        "complete": p.emerald,
        "incomplete": p.saffron,
    }.get(status, p.ink_secondary)


def status_label(status: str) -> str:
    from .l10n import tr
    return {
        "pending": tr("待读", "Pending"),
        "L1": tr("已读 L1", "Read · L1"),
        "L2": "L2",
        "complete": tr("已蒸馏", "Distilled"),
        "incomplete": tr("不完整", "Incomplete"),
    }.get(status, status)


def with_alpha(hex_color: str, alpha: float) -> str:
    """#RRGGBB + alpha → rgba() string for QSS."""
    h = hex_color.lstrip("#")
    r, g, b = int(h[0:2], 16), int(h[2:4], 16), int(h[4:6], 16)
    return f"rgba({r},{g},{b},{int(alpha * 255)})"


def build_qss(p: Palette) -> str:
    return f"""
QMainWindow, QDialog {{ background: {p.canvas}; }}
QWidget {{ color: {p.ink}; font-size: 13px; }}
QLabel {{ background: transparent; }}

QListWidget#sidebar {{
    background: {p.sidebar}; border: none; outline: none; padding: 6px 4px;
}}
QListWidget#sidebar::item {{
    color: {p.ink_secondary}; border-radius: 6px; padding: 7px 10px; margin: 1px 4px;
}}
QListWidget#sidebar::item:hover {{ background: {with_alpha(p.ink, 0.05)}; }}
QListWidget#sidebar::item:selected {{ background: {p.accent}; color: {"#F4F5F0" if not p.is_dark else "#101820"}; }}

QListWidget#contentList {{
    background: {p.card}; border: 1px solid {p.hairline}; border-radius: 8px; outline: none; padding: 3px;
}}
QListWidget#contentList::item {{ border-radius: 5px; padding: 7px 8px; color: {p.ink}; }}
QListWidget#contentList::item:hover {{ background: {with_alpha(p.accent, 0.06)}; }}
QListWidget#contentList::item:selected {{ background: {with_alpha(p.accent, 0.14)}; color: {p.ink}; }}

QFrame#card {{
    background: {p.card}; border: 1px solid {p.hairline}; border-radius: 10px;
}}
QFrame#well {{
    background: {p.well}; border: 1px solid {p.hairline}; border-radius: 8px;
}}
QFrame#banner {{
    background: {with_alpha(p.saffron, 0.10)}; border: 1px solid {with_alpha(p.saffron, 0.45)};
    border-radius: 8px;
}}

QPushButton {{
    background: {p.card}; color: {p.ink}; border: 1px solid {p.hairline};
    border-radius: 6px; padding: 5px 14px;
}}
QPushButton:hover {{ border-color: {p.accent}; color: {p.accent}; }}
QPushButton:pressed {{ background: {with_alpha(p.accent, 0.10)}; }}
QPushButton:disabled {{ color: {with_alpha(p.ink_secondary, 0.55)}; border-color: {with_alpha(p.hairline, 0.6)}; }}
QPushButton#primary {{
    background: {p.accent}; color: {"#F4F5F0" if not p.is_dark else "#101820"}; border: none; font-weight: 600;
}}
QPushButton#primary:hover {{ background: {p.navy if not p.is_dark else "#8FB5D8"}; color: {"#F4F5F0" if not p.is_dark else "#101820"}; }}
QPushButton#primary:disabled {{ background: {with_alpha(p.accent, 0.4)}; }}
QPushButton#linkish {{ background: transparent; border: none; color: {p.accent}; padding: 1px 4px; text-align: left; }}
QPushButton#linkish:hover {{ text-decoration: underline; }}

QLineEdit, QTextEdit, QPlainTextEdit, QSpinBox, QDoubleSpinBox, QComboBox {{
    background: {p.card}; color: {p.ink}; border: 1px solid {p.hairline};
    border-radius: 6px; padding: 5px 8px; selection-background-color: {with_alpha(p.accent, 0.35)};
}}
QLineEdit:focus, QTextEdit:focus, QSpinBox:focus, QDoubleSpinBox:focus, QComboBox:focus {{
    border-color: {p.accent};
}}
QComboBox::drop-down {{ border: none; width: 22px; }}
QComboBox QAbstractItemView {{
    background: {p.card}; color: {p.ink}; border: 1px solid {p.hairline};
    selection-background-color: {with_alpha(p.accent, 0.16)}; selection-color: {p.ink};
}}

QTextBrowser {{
    background: {p.card}; color: {p.ink}; border: 1px solid {p.hairline};
    border-radius: 8px; padding: 10px;
}}

QSplitter::handle {{ background: transparent; width: 6px; }}
QScrollArea {{ border: none; background: transparent; }}
QScrollArea > QWidget > QWidget {{ background: transparent; }}
QScrollBar:vertical {{ background: transparent; width: 10px; margin: 2px; }}
QScrollBar::handle:vertical {{ background: {with_alpha(p.ink_secondary, 0.35)}; border-radius: 4px; min-height: 30px; }}
QScrollBar::add-line, QScrollBar::sub-line {{ height: 0; width: 0; }}
QScrollBar:horizontal {{ background: transparent; height: 10px; margin: 2px; }}
QScrollBar::handle:horizontal {{ background: {with_alpha(p.ink_secondary, 0.35)}; border-radius: 4px; min-width: 30px; }}

QTabWidget::pane {{ border: none; top: 6px; }}
QTabBar::tab {{
    background: transparent; color: {p.ink_secondary}; padding: 6px 14px;
    border: 1px solid {p.hairline}; border-radius: 6px; margin-right: 6px;
}}
QTabBar::tab:selected {{ background: {p.accent}; color: {"#F4F5F0" if not p.is_dark else "#101820"}; border-color: {p.accent}; }}

QStatusBar {{ background: {p.sidebar}; color: {p.ink_secondary}; }}
QDockWidget {{ color: {p.ink}; titlebar-close-icon: none; }}
QToolTip {{ background: {p.card}; color: {p.ink}; border: 1px solid {p.hairline}; padding: 4px 6px; }}
"""
